### DISCUSSION TITLE
Can threeagent be used with react-native?

### DISCUSSION THREAD

**Discussion by hkjels** [DougHamil/threeagent#45] [Q&A] [answered]
I believe `three.js` is possible to use with `react-native`, but I guess the `render`-method of `threeagent` will have to be changed right?

**Comment by DougHamil** [comment, marked as answer]
I haven't used react-native before, but I started digging into it yesterday. You're right, it looks like we just need to allow a ThreeJS `Renderer` instance to be passed into threeagent's `render` fn.

Currently, threeagent creates the renderer and sets up the canvas, but this is just for convenience, not necessity. I have a [working example](https://github.com/DougHamil/threeagent/blob/react-native-support/examples-react-native/src/app.cljs) and I'll get the changes merged into the latest version after I do a bit more testing.

The only issues that came up were from bugs in the latest version of `three` + `expo-three`, but threeagent works fine once you get a `expo-three` Renderer instance created.

**Reply by hkjels** [reply]
Sounds good
